src/main/java/dev/morling/onebrc/WeatherStationFactory.java

```diff
@@ -0,0 +1,148 @@
+                if (count == -1) {
+                    // TODO: Reset nameSource and read instead of throwing?
+                    throw new Exception("Name source exhausted");
+                }
+
+                var nameBuf = new StringBuilder(nameLen);
+                nameBuf.append(buf, 0, nameLen);
+
+                if (Character.isWhitespace(nameBuf.charAt(0))) {
+                    nameBuf.setCharAt(0, readNonSpace(nameSource));
+                }
+                if (Character.isWhitespace(nameBuf.charAt(nameBuf.length() - 1))) {
+                    nameBuf.setCharAt(nameBuf.length() - 1, readNonSpace(nameSource));
+                }
+
+                var name = nameBuf.toString();
+                int nameByteLen = name.getBytes(StandardCharsets.UTF_8).length;
+
+                while (names.contains(name) || nameByteLen > 100) {
+                    // If the character length is over 100
+                    if (nameByteLen > 100) {
+                        nameBuf.deleteCharAt(nameBuf.length() - 1);
+                        if (Character.isWhitespace(nameBuf.charAt(nameBuf.length() - 1))) {
+                            nameBuf.setCharAt(nameBuf.length() - 1, readNonSpace(nameSource));
+                        }
+                    }
+                    // Else: name is not unique
+                    else {
+                        nameBuf.setCharAt(rnd.nextInt(nameBuf.length()), readNonSpace(nameSource));
+                    }
+
+                    name = nameBuf.toString();
+                    nameByteLen = name.getBytes(StandardCharsets.UTF_8).length;
+                }
+
+                if (name.indexOf(';') != -1) {
+                    throw new Exception("Station name contains a semicolon!");
+                }
+
+                names.add(name);
+                minLen = Integer.min(minLen, nameByteLen);
+                maxLen = Integer.max(maxLen, nameByteLen);
+                var lat = Double.parseDouble(row.substring(row.indexOf(';') + 1));
+                // Guesstimate mean temperature using cosine of latitude
+                var avgTemp = (double) (30 * Math.cos(Math.toRadians(lat))) - 10;
+                weatherStations.add(new WeatherStation(seed, name, avgTemp));
+            }
+        }
+        System.out.format("Generated %,d station names with length from %,d to %,d%n", KEYSET_SIZE, minLen, maxLen);
+        return weatherStations;
+    }
+
+    private static void skipComments(BufferedReader rows) throws IOException {
+        while (rows.readLine().startsWith("#")) {
+        }
+    }
+
+    private static char readNonSpace(StringReader nameSource) throws Exception {
+        char c;
+        do {
+            var n = nameSource.read();
+            if (n == -1) {
+                // TODO: Reset nameSource and read instead of throwing?
+                throw new Exception("Name source exhausted");
+            }
+            c = (char) n;
+        } while (c == ' ');
```

Comment:
As an OCD moment, I'd like to change this to `while (Character.isWhitespace(c))` to align with the other tests.

Reply:
Good catch! Updated